torch/csrc/lazy/core/hash.h

```diff
@@ -74,7 +74,18 @@ static inline hash_t Hash(const c10::ScalarType& value) {
 }
 
 static inline hash_t Hash(const c10::Scalar& value) {
-  return DataHash(&value, sizeof(value));
+  switch(value.type()){
+  case c10::ScalarType::ComplexDouble:
+    return Hash(value.toComplexDouble());
+  case c10::ScalarType::Double:
+    return Hash(value.toDouble());
+  case c10::ScalarType::Long:
+    return Hash(value.toLong());
+  case c10::ScalarType::Bool:
+    return Hash(value.toBool());
+  default:
+    throw std::runtime_error("Unknown scalar type.");
```

Comment:
Maybe better to use TORCH_CHECK if we want users to know the error or TORCH_INTERNAL_ASSERT.